src/main.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+  res.setHeader('Content-type', 'application/json');
+
+  const paramsSplitted = req.url.slice(1).split('?');
+  const paths = paramsSplitted[0].split('/');
+  const params = new URLSearchParams(paramsSplitted[1]);
+  const errors = [];
+  const query = Object.fromEntries(new URLSearchParams(params));
+
+  if (errors.length) {
+    res.statusCode = 400;
+    res.statusMessage = 'Bad request';
+    res.end(JSON.stringify({ errors }));
+
+    return;
+  }
+
+  const result = {
+    parts: paths,
+    query,
+  };
+
+  res.end(JSON.stringify(result));
+});
+
+server.listen(8080);
```

Comment:
You have an unused `app.js` file in this repo. Remove the `.listen` call from here, import your server in `app.js` and call `.listen` there.
Also, in this case, **this** file should be called `server.js` instead of `main.js`.

This way, you will have a separate module as an entrypoint for your app and separate module for the server implementation.


```suggestion
```

Reply:
Also, it's a good idea to add a callback to `.listen` call like this:

```
server.listen(port, () => {
  console.log(`Server is listening at ${port}.`);
});
```
This shows that server is indeed running.